lib/howl/interactions/file_selection.moon

```diff
@@ -111,23 +111,39 @@ class FileSelector
       @command_line.text = path.basename
     self.finish path
 
+  _open: =>
+    app.editor\cancel_preview!
+    file = @list_widget.selection and @list_widget.selection.file
+    name = @list_widget.selection and @list_widget.selection.name
+    if not @opts.allow_new and (not file or not file.exists)
+      log.error 'Invalid path: '..tostring(file)
+      return
+
+    if name == ".#{separator}"
+      @_submit @directory
+      return
+
+    if file.exists and file.is_directory
+      @_chdir file
+    else
+      @_submit file
+
   keymap:
     enter: =>
-      app.editor\cancel_preview!
-      file = @list_widget.selection and @list_widget.selection.file
-      name = @list_widget.selection and @list_widget.selection.name
-      if not @opts.allow_new and (not file or not file.exists)
-        log.error 'Invalid path: '..tostring(file)
-        return
-
-      if name == ".#{separator}"
-        @_submit @directory
-        return
-
-      if file.exists and file.is_directory
-        @_chdir file
+      @_open!
+
+    tab: =>
+      count = 0
+      for item in *@list_widget.items
+        count += 1 unless item.is_new
+      if count == 1
+        -- open a unique completion
+        if @list_widget.selection and not @list_widget.selection.is_new
+          @_open!
       else
-        @_submit file
+        -- cycle non-new entries
+        @list_widget\select_next! unless @list_widget.selection.is_new
```

Comment:
If you press tab whilst on the New entry, it would advance two positions otherwise. Hopefully an updated comment will clarify.